### PR TITLE
[mqtt] Deliver peer certificate to authorizer

### PR DESCRIPTION
**Pull request by dmolokanov**
In order to retrieve a peer certificate for authentication purposes we need a newer version of tokio wrapper around native-tls which is not released yet.
* Switched to use `tokio-native-tls` instead outdated `tokio-tls`
* Add code to extract certificate and pass it to authentication logic

### CODE REVIEW

File: mqtt/mqtt-broker/src/transport.rs [hunk 1]
```diff
@@ -239,7 +239,31 @@ impl StreamSelector {
     pub fn peer_addr(&self) -> std::io::Result<SocketAddr> {
         match self {
             StreamSelector::Tcp(stream) => stream.peer_addr(),
-            StreamSelector::Tls(stream) => stream.get_ref().peer_addr(),
+            StreamSelector::Tls(stream) => stream.get_ref().get_ref().get_ref().peer_addr(),
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker/src/transport.rs:242]: 0_o, wow

File: mqtt/mqtt-broker/src/connection.rs [hunk 2]
```diff
@@ -73,10 +74,10 @@ pub async fn process<I>(
     mut broker_handle: BrokerHandle,
 ) -> Result<(), Error>
 where
-    I: AsyncRead + AsyncWrite + Unpin,
+    I: AsyncRead + AsyncWrite + GetPeerCertificate<Certificate = Certificate> + Unpin,
 {
     // TODO read certificate from stream when ready
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker/src/connection.rs:79]: todo is no longer needed?

File: mqtt/mqtt-broker/src/transport.rs (hunk 1 above)
Comment (dmolokanov) [reply mqtt/mqtt-broker/src/transport.rs:242]: yep, not very pleasant API

File: mqtt/mqtt-broker/src/transport.rs [hunk 3]
```diff
@@ -239,7 +239,31 @@ impl StreamSelector {
...
+    }
+}
+
+pub trait GetPeerCertificate {
+    type Certificate;
+
+    fn peer_certificate(&self) -> Result<Option<Self::Certificate>, Error>;
+}
+
+impl GetPeerCertificate for StreamSelector {
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker/src/transport.rs:253]: is it possible to unit test?

Comment (dmolokanov) [reply mqtt/mqtt-broker/src/transport.rs:253]: what exact aspect you want to check?

File: mqtt/mqtt-broker/src/connection.rs (hunk 2 above)
Comment (lfitchett) [reply mqtt/mqtt-broker/src/connection.rs:79]: Looks like todo is still here

**Pull request merged**